### PR TITLE
[G2M] Navbar - Add default & Custom Styling

### PR DESCRIPTION
**Pull request by Hamaad102**
https://www.notion.so/eqproduct/Lumen-Labs-Navbar-e27871f35adf4462924865cd7f3bdaca?pvs=4

**Changes**
- Added navbar component
- Added two stories, one with the unstyled navbar and one styled

### CODE REVIEW

File: src/components/navbar/navbar-item.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        </div>
+        <div className={`${isSelected ? NavbarItemClasses.textActive : NavbarItemClasses.text}`}>{label}</div>
+      </div>
+    </div>
+  )
+})
+
+NavbarItem.propTypes = {
+  id: PropTypes.string,
+  classes: PropTypes.object,
```
Comment (geoerika) [inline src/components/navbar/navbar-item.js:44]: I would extend `propTypes` for `classes` to the inside keys:
```
classes: PropTypes.shape({
  itemContainer: PropTypes.string,
  ...
```

File: src/components/navbar/index.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+      className={navbarClasses.root} 
+      {...rest}
+    >
+      {React.Children.map(children, (child) => {
+        if (React.isValidElement(child) && child.type.__docgenInfo.displayName === 'NavbarHeader') {
+          return React.cloneElement(child)
+        }
+      })}
+
+      <div className="navbar__items flex items-center justify-center">
```
Comment (geoerika) [inline src/components/navbar/index.js:24]: I would add here a `navbarClasses.items` class so we can customize this as we wish.

**Review by geoerika — APPROVED** [review]
Overall it looks good, but would do a few changes for clarity. See my comments.

**Review by amy83762100 — APPROVED** [review]

File: src/components/navbar/navbar-item.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  label: PropTypes.string,
+  icon: PropTypes.node.isRequired,
+  onClick: PropTypes.func,
+  isSelected: PropTypes.bool,
+}
+
+NavbarItem.defaultProps = {
```
Comment (geoerika) [inline src/components/navbar/navbar-item.js:51]: @Hamaad102 you moved propTypes for classes keys to defaultProps ...

File: src/components/navbar/index.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+          return React.cloneElement(child)
+        }
+      })}
+    </div>
+  )
+})
+
+Navbar.propTypes = {
+  children: PropTypes.node.isRequired,
+  classes: PropTypes.object,
```
Comment (geoerika) [inline src/components/navbar/index.js:48]: Please, make the same changes to all classes in all components for propTypes.

File: src/components/navbar/navbar-item.js (hunk 3 above)
Comment (Hamaad102) [reply src/components/navbar/navbar-item.js]: Sorry,  this was a brain fart

File: src/components/navbar/navbar-item.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+  ...rest
+}, ref) => {
+  const NavbarItemClasses = Object.freeze({
+    itemContainer: `navbar-item__main-container ${classes.itemContainer} cursor-pointer mx-6`,
+    itemContainerActive: `navbar-item__main-container--active ${classes.itemContainerActive} py-2 px-4 mx-2 rounded-full bg-black`,
+    contentContainer: `navbar-item__content-container ${classes.contentContainer} flex flex-row items-center`,
+    icon: `navbar-item__icon ${classes.icon} pr-2.5`,
+    iconActive: `navbar-item__icon ${classes.iconActive} pr-2.5 text-white`,
+    text: `navbar-item__text ${classes.text} text-black`,
+    textActive: `navbar-item__text ${classes.textActive} text-white`,
```
Comment (kc-leung) [inline src/components/navbar/navbar-item.js:21]: looks good lets add active for active classes so user can target then individually. navbar-item__icon-active

**Review by kc-leung — APPROVED** [review]

**Pull request merged**